presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/PrestoClusterManager.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.rubix;
+
+import com.google.common.collect.ImmutableList;
+import com.qubole.rubix.spi.ClusterManager;
+import com.qubole.rubix.spi.ClusterType;
+import io.prestosql.spi.Node;
+import io.prestosql.spi.NodeManager;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.qubole.rubix.spi.ClusterType.PRESTOSQL_CLUSTER_MANAGER;
+import static java.util.Objects.requireNonNull;
+
+public class PrestoClusterManager
+        extends ClusterManager
+{
+    private static NodeManager nodeManager;
```

Comment:
Make this thread safe. make `setNodeManager` synchronized and add synchronized getter.